GG/GG/Font.h

```diff
@@ -206,6 +209,22 @@ class GG_API Font
 
         virtual ~TextElement(); ///< Virtual dtor.
 
+        /** Rebind to \p whole_text, by binding the SubString data member text to \p whole_text.
+
+            Binding to a new \p whole_text is very fast compared to re-parsing the entire \p
+            whole_text and allows TextElements of TextElementType TEXT to be changed quickly if it
+            is known that the parse would be the same.
+
+            One use case is copying or copy constructing a TextControl.
+
+            Another use case is constructing a string and TextElement vector pair with
```

Comment:
I don't mind the name Bind.

But it's not great when the comments explaining the purpose / result of the function use the functions name to explain itself... ie. saying Bind "rebinds" something doesn't help much. The rest of the comment is better, though.

The example aren't very clear, though... More detail would be better, about why situations would arise in which you'd want to change the pointed to string.

After reading this several times, I think it would make sense when you want to do a text parse or layout once, and then create several different controls that have the same text layout and text contents, but each of them needs to keep there own internal copy of the text. So, while the pointer diffs are all the same, since the text contents in the same, the pointer to the string in each TextElement (and its contained Substring) needs to be set to the appropriate copy of the string.

At this point though, I wonder why you need to index into the strings, rather than just keeping the necessary string inside each TextElement itself? If you end up making another copy of the whole string every time you make another copy of the control, then what's the benefit of indexing instead of having a bunch of smaller strings internally...?